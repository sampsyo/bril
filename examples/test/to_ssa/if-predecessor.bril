@main() {
.entry:
    cond: bool = const true;
    b: int = const 42;
    br cond .true .false;
.true:
    # `b` is defined in predecessor `.entry`, so it should be passed along to the phi in `.zexit`.
    jmp .zexit;
.false:
    b: int = const 1;
    jmp .zexit;
.zexit:
    print b;
}
