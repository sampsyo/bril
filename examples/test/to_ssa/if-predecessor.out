@main {
.entry:
  cond.0: bool = const true;
  b.0: int = const 42;
  br cond.0 .true .false;
.true:
  jmp .zexit;
.false:
  b.1: int = const 1;
  jmp .zexit;
.zexit:
  b.2: int = phi b.1 b.0 .false .true;
  print b.2;
  ret;
}
